lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -9639,6 +9639,82 @@ namespace Js
         return superCtor;
     }
 
+    Var JavascriptOperators::OP_ImportCall(Var specifier, ScriptContext* scriptContext)
+    {
+        ModuleRecordBase *moduleRecordBase = nullptr;
+        SourceTextModuleRecord *moduleRecord = nullptr;
+
+        JavascriptFunction* pfuncCaller;
+        if (JavascriptStackWalker::GetCaller(&pfuncCaller, scriptContext) && pfuncCaller && pfuncCaller->IsScriptFunction())
+        {
+            FunctionBody* parentFuncBody = pfuncCaller->GetFunctionBody();
+            ModuleID id = parentFuncBody->GetModuleID();
+
+            LPCOLESTR moduleName = Js::JavascriptString::FromVar(specifier)->GetSz();
+
+            HRESULT hr = 0;
+            if (id != kmodGlobal)
+            {
+                SourceTextModuleRecord *referenceModuleRecord = parentFuncBody->GetScriptContext()->GetLibrary()->GetModuleRecord(id);
+
+                ASYNC_HOST_OPERATION_START(scriptContext->GetThreadContext());
+                hr = scriptContext->GetHostScriptContext()->FetchImportedModule(referenceModuleRecord, moduleName, &moduleRecordBase);
+                ASYNC_HOST_OPERATION_END(scriptContext->GetThreadContext());
+            }
+            else
+            {
+                DWORD_PTR dwReferencingSourceContext = parentFuncBody->GetFunctionInfo()->GetFunctionProxy()->GetSourceContextInfo()->dwHostSourceContext;
+
+                ASYNC_HOST_OPERATION_START(scriptContext->GetThreadContext());
+                hr = scriptContext->GetHostScriptContext()->FetchImportedModuleFromScript(dwReferencingSourceContext, moduleName, &moduleRecordBase);
+                ASYNC_HOST_OPERATION_END(scriptContext->GetThreadContext());
+            }
+
+            if (FAILED(hr))
+            {
+                Js::JavascriptError * error = scriptContext->GetLibrary()->CreateURIError();
+                Var value = JavascriptString::NewCopySz(moduleName, scriptContext);
+                JavascriptOperators::OP_SetProperty(error, PropertyIds::message, value, scriptContext);
```

Comment:
Updated in 2nd iteration.

---
In reply to: [114844985](https://github.com/Microsoft/ChakraCore/pull/2913#discussion_r114844985) [](ancestors = 114844985)